### ISSUE TITLE
unstable_Pagination does not fire onChange for next and previous buttons

### ISSUE THREAD

**Issue by tusharcoepit** [carbon-design-system/ibm-security#1051]
Pagination component has a performance issue when data is huge.

https://github.com/carbon-design-system/ibm-security/issues/162

As an alternative we are trying to use `unstable_Pagination`, but it does not trigger onChange on click of next and previous buttons.

Any other alternatives to handle the performance issue with Pagination is also welcome.

**Comment by SimonFinney** [comment]
Thanks @tusharcoepit - are you leveraging the component per the [example story in the Carbon repo](https://www.github.com/carbon-design-system/carbon/tree/main/packages%2Freact%2Fsrc%2Fcomponents%2FPagination%2Fexperimental%2FPagination-story.js)?

**Comment by tusharcoepit** [comment]
Yes @SimonFinney, we are referencing the storybook. In-fact on story book as well we don't see onChange event triggering for next / previous buttons and it triggers only on page selection from dropdown. 

We are referring this -> https://react.carbondesignsystem.com/?path=/story/experimental-unstable-pagination--with-a-page-selector

**Comment by SimonFinney** [comment]
> Yes @SimonFinney, we are referencing the storybook. In-fact on story book as well we don't see onChange event triggering for next / previous buttons and it triggers only on page selection from dropdown.
> 
> We are referring this -> https://react.carbondesignsystem.com/?path=/story/experimental-unstable-pagination--with-a-page-selector

Thanks for clarifying! I believe this issue should be [created in Carbon](https://github.com/carbon-design-system/carbon/issues/new/choose) as the component is proxied from there.

**Comment by tusharcoepit** [comment]
Created issue in carbon https://github.com/carbon-design-system/carbon/issues/9313
